pkg/network/network.go

```diff
@@ -271,39 +267,13 @@ func ParseAddr(addr string) (net.IP, *net.IPNet, error) {
 	return ip, network, err
 }
 
-// reserveSubnet returns if subnet was successfully reserved for given period:
-//   - false, if it already has unexpired reservation
-//   - true, if new reservation was created or expired one renewed
-//
-// uses sync.Map to manage reservations thread-safe
-var reserveSubnet = func(subnet string, period time.Duration) bool {
-	// put 'zero' reservation{} Map value for subnet Map key
-	// to block other processes from concurrently changing this subnet
-	zero := reservation{}
-	r, loaded := reservedSubnets.LoadOrStore(subnet, zero)
-	// check if there was previously issued reservation
-	if loaded {
-		// back off if previous reservation was already set to 'zero'
-		// as then other process is already managing this subnet concurrently
-		if r == zero {
-			klog.Infof("backing off reserving subnet %s (other process is managing it!): %+v", subnet, &reservedSubnets)
-			return false
-		}
-		// check if previous reservation expired
-		createdAt := r.(reservation).createdAt
-		if time.Since(createdAt) < period {
-			// unexpired reservation: restore original createdAt value
-			reservedSubnets.Store(subnet, reservation{createdAt: createdAt})
-			klog.Infof("skipping subnet %s that has unexpired reservation: %+v", subnet, &reservedSubnets)
-			return false
-		}
-		// expired reservation: renew setting createdAt to now
-		reservedSubnets.Store(subnet, reservation{createdAt: time.Now()})
-		klog.Infof("reusing subnet %s that has expired reservation: %+v", subnet, &reservedSubnets)
-		return true
+// reserveSubnet returns releaser if subnet was successfully reserved for given period, creating lock for subnet to avoid race condition between multiple minikube instances (especially white testing in parallel).
```

Comment:
```suggestion
// reserveSubnet returns releaser if subnet was successfully reserved for given period, creating lock for subnet to avoid race condition between multiple minikube instances (especially while testing in parallel).
```

Reply:
eagle-eye :)